File: Sprint 4/Assets/Script/GUI/healthbar/healthbar.js
var healthBarBackground: Texture; //the image for the background of the entire healthbar (the wood look with gears)
var healthBackground: Texture; //this is what the actual health meter should look like when empty (red)
var healthForeground: Texture; //this is what the actual health meter should look like when full (green)
enum healthBarLocationValues {topLeft, topCenter, topRight, middleLeft, middleCenter, middleRight, bottomLeft, bottomCenter, bottomRight, custom}
var healthBarLocation : healthBarLocationValues = healthBarLocationValues.bottomLeft;

var healthBarScale = 0.3;
var healthBarSizePercent = 15;

var maxHealth = 100;//max health player can have
var curHealth = 100;//current health player has

private var healthBarWidth : float;
private var healthBarHeight : float;
private var healthBarCenter : Vector2;
var healthBarCenterCustom : Vector2;


private var healthWidth : float = -35;
private var healthHeight: float = -120;
private var healthLocationX = 377;
private var healthLocationY = 635;


function Start () {
	setHealthBarLocation();	
}

function OnGUI () {
 	//GUI.DrawTexture(Rect(healthBarCenter.x - healthBarWidth/2,healthBarCenter.y-healthBarHeight/2,healthBarWidth,healthBarHeight),healthBarBackground);
 	GUI.DrawTexture(Rect(287, 500, 150, 150), healthBarBackground);
 	GUI.DrawTexture(Rect(healthLocationX, healthLocationY, healthWidth, healthHeight), healthBackground); 
 	GUI.DrawTexture(Rect(healthLocationX, healthLocationY, healthWidth, healthHeight/(maxHealth/curHealth)), healthForeground); 
}

function setHealthBarLocation () {
	healthBarWidth = Screen.width*healthBarSizePercent/100.0;
	healthBarHeight = healthBarWidth;

	//sets mapCenter based on enum selection
	if(healthBarLocation == healthBarLocationValues.topLeft){
		healthBarCenter = Vector2(healthBarWidth/2, healthBarHeight/2);
	} else if(healthBarLocation == healthBarLocationValues.topCenter){
		healthBarCenter = Vector2(Screen.width/2, healthBarHeight/2);
	} else if(healthBarLocation == healthBarLocationValues.topRight){
		healthBarCenter = Vector2(Screen.width-healthBarWidth/2, healthBarHeight/2);
	} else if(healthBarLocation == healthBarLocationValues.middleLeft){
		healthBarCenter = Vector2(healthBarWidth/2, Screen.height/2);
	} else if(healthBarLocation == healthBarLocationValues.middleCenter){
		healthBarCenter = Vector2(Screen.width/2, Screen.height/2);
	} else if(healthBarLocation == healthBarLocationValues.middleRight){
		healthBarCenter = Vector2(Screen.width-healthBarWidth/2, Screen.height/2);
	} else if(healthBarLocation == healthBarLocationValues.bottomLeft){
		healthBarCenter = Vector2(healthBarWidth/2, Screen.height - healthBarHeight/2);
	} else if(healthBarLocation == healthBarLocationValues.bottomCenter){
		healthBarCenter = Vector2(Screen.width/2, Screen.height - healthBarHeight/2);
	} else if(healthBarLocation == healthBarLocationValues.bottomRight){
		healthBarCenter = Vector2(Screen.width-healthBarWidth/2, Screen.height - healthBarHeight/2);
	} else if(healthBarLocation == healthBarLocationValues.custom){
		healthBarCenter = healthBarCenterCustom;
	}
} 

function Update()
{
}

function OnTriggerEnter(obj: Collider)
{
	if(obj.getObjectByTag == "building")
	{
		print("boobies");
		if(curHealth > 1)
		{
			curHealth--;
		}
	}
}

File: Sprint 4/Assets/Script/GUI/healthbar/_Scripts/HealthBar.cs
// Author: David Preischel, Rich Canada
// Edited: 11/13/13 Added go back to main menu when dead - David Preischel

using UnityEngine;
using System.Collections;



public class HealthBar : MonoBehaviour {

    //Build Texture objects for the health bar.
   public Texture healthBarBackground; //the image for the background of the entire healthbar (the wood look with gears)
   public Texture healthBackground; //this is what the actual health meter should look like when empty (red)
   public Texture healthForeground; //this is what the actual health meter should look like when full (green)

    //Positioning enum values
    public enum healthBarLocationValues { topLeft, topCenter, topRight, middleLeft, middleCenter, middleRight, bottomLeft, bottomCenter, bottomRight, custom }
    public healthBarLocationValues healthBarLocation = healthBarLocationValues.bottomLeft;
    private float healthBarWidth;
    private float healthBarHeight;
    private Vector2 healthBarCenter;
    public Vector2 healthBarCenterCustom;


    public float healthBarScale = 0.3f;
    public int healthBarSizePercent = 15;

    public int maxHealth = 100;//max health player can have
    public int curHealth = 100;//current health player has

    private float healthWidth = -35;
    private float healthHeight = -120;
    private int healthLocationX = 377;
    private int healthLocationY = 635;

	// Use this for initialization
	void Start () {
        setHealthBarLocation();	
	}

    //Positions the Health Bar based on the location chosen via the enum
    void setHealthBarLocation(){
        healthBarWidth = Screen.width * healthBarSizePercent / 100.0f;
        healthBarHeight = healthBarWidth;

        //sets mapCenter based on enum selection
        if (healthBarLocation == healthBarLocationValues.topLeft)
        {
            healthBarCenter = new Vector2(healthBarWidth / 2, healthBarHeight / 2);
        }
        else if (healthBarLocation == healthBarLocationValues.topCenter)
        {
            healthBarCenter = new Vector2(Screen.width / 2, healthBarHeight / 2);
        }
        else if (healthBarLocation == healthBarLocationValues.topRight)
        {
            healthBarCenter = new Vector2(Screen.width - healthBarWidth / 2, healthBarHeight / 2);
        }
        else if (healthBarLocation == healthBarLocationValues.middleLeft)
        {
            healthBarCenter = new Vector2(healthBarWidth / 2, Screen.height / 2);
        }
        else if (healthBarLocation == healthBarLocationValues.middleCenter)
        {
            healthBarCenter = new Vector2(Screen.width / 2, Screen.height / 2);
        }
        else if (healthBarLocation == healthBarLocationValues.middleRight)
        {
            healthBarCenter = new Vector2(Screen.width - healthBarWidth / 2, Screen.height / 2);
        }
        else if (healthBarLocation == healthBarLocationValues.bottomLeft)
        {
            healthBarCenter = new Vector2(healthBarWidth / 2, Screen.height - healthBarHeight / 2);
        }
        else if (healthBarLocation == healthBarLocationValues.bottomCenter)
        {
            healthBarCenter = new Vector2(Screen.width / 2, Screen.height - healthBarHeight / 2);
        }
        else if (healthBarLocation == healthBarLocationValues.bottomRight)
        {
            healthBarCenter = new Vector2(Screen.width - healthBarWidth / 2, Screen.height - healthBarHeight / 2);
        }
        else if (healthBarLocation == healthBarLocationValues.custom)
        {
            healthBarCenter = healthBarCenterCustom;
        }
    }
    //Drawing the Health bar to the HUD
    void OnGUI(){
        GUI.DrawTexture(new Rect(287, 500, 150, 150), healthBarBackground);//Cool looking background
        GUI.DrawTexture(new Rect(healthLocationX, healthLocationY, healthWidth, healthHeight), healthBackground);//Red Bar
        GUI.DrawTexture(new Rect(healthLocationX, healthLocationY, healthWidth, (healthHeight *curHealth) / maxHealth), healthForeground);//Green Bar
    }
	
	// Update is called once per frame
	void Update () {

        //Check for Collisions - Subtract health when a building is bumped
		if(curHealth <= 0){
			//Do shit, end game
            PlayerPrefs.SetInt("Level", 1);
            Application.LoadLevel("mainmenu");
		}
		print (curHealth);
		
	}
		
	void OnCollisionEnter(Collision col)
	{
		if(col.gameObject.tag == "building")
		{
			print ("colliding");
			if (curHealth > 1)
        	{
				curHealth -= 25;
			}
		}
	}
}
